socorro/collector/wsgi_breakpad_collector.py

```diff
@@ -84,8 +87,32 @@ def _make_raw_crash_and_dumps(self, form):
 
     #--------------------------------------------------------------------------
     def POST(self, *args):
+        # Handle gzipped form posts
+        if web.ctx.env.get('HTTP_CONTENT_ENCODING') == 'gzip':
+            gzip_header = 16 + zlib.MAX_WBITS
+            data = zlib.decompress(web.webapi.data(), gzip_header)
+            fp = cStringIO.StringIO(data)
+            e = web.ctx.env.copy()
+
+            # this is how web.webapi.rawinput() handles
+            # multipart/form-data, as of this writing
+            fs = cgi.FieldStorage(fp=fp, environ=e, keep_blank_values=1)
+            fsdict = dict([(k, fs[k]) for k in fs.keys()])
+            def process_fieldstorage(fs):
```

Comment:
this method is deceptive.  Embedded here it looks like a closure.  However, the variable ```fs``` is a parameter rather than the ```fs``` value from the outer scope.